codice/ddf
# DDF-2413 Refactored CSW to use new taxonomy and cleaned up code

#### What does this PR do?
This PR refactors the entire CSW project.  It:
- aligns everything to the new taxonomy
- removes dead and unused code
- java 8ifies code

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@glenhein @jrnorth 
#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@jlcsmith
@kcwire
#### How should this be tested?
Build / Install DDF.  Configure a CSW Federated source and observe correct behavior.  Perform CSW operations (create / update / delete) and query on the CSW Endpoint and observer proper behavior.
#### Any background context you want to provide?
#### What are the relevant tickets?
https://codice.atlassian.net/browse/DDF-2413
#### Screenshots (if appropriate)
#### Checklist:
